File: 3429-special-array-i/3429-special-array-i.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isArraySpecial = function(nums) {
    
    var odd = nums[0]%2!=0;
   for (let i = 1; i < nums.length; i++) {
    if ((odd && nums[i] % 2 !== 0) || (!odd && nums[i] % 2 === 0)) {
            return false;
        }
        odd = !odd;
    }

    return true;
};

File: 3429-special-array-i/3429-special-array-i.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        
        bool odd = nums[0]%2== 0 ? false : true;

        for(int i=1; i<nums.size() ; i++){
            if((odd && nums[i]%2!=0 )|| (!odd && nums[i]%2==0))
            return false;

            else {
                odd = !odd;
            }
        }
        return true;
    }
};